build-system/tasks/extension-helpers.js

```diff
@@ -127,7 +129,13 @@ function declareExtension(name, version, options, extensionsObject) {
   const defaultOptions = {hasCss: false, npm: undefined};
   const versions = Array.isArray(version) ? version : [version];
   versions.forEach((v) => {
-    extensionsObject[`${name}-${v}`] = {
+    // If a destName is given, make it as a part of the key as it is
+    // most likely needed to make the entry unique for instances where
+    // multiple entries share the same "entryPoint/name"  but have different
+    // destination name. This allows for a 1 to many relationship between
+    // entryPoint and output (1 -> *).
+    const key = options ? options.destName ?? name : name;
```

Comment:
Curious if this would work:
```suggestion
    const key = options?.destName ?? name;
```

Reply:
seems like it does. updating